rest_framework_jwt/models.py

```diff
@@ -14,3 +17,8 @@ class Device(models.Model):
     name = models.CharField(_('Device name'), max_length=255)
     details = models.CharField(_('Device details'), max_length=255, blank=True)
     last_request_datetime = models.DateTimeField(auto_now=True)
+
+    @classmethod
+    def create(cls, *args, **kwargs):
+        permanent_token = binascii.hexlify(os.urandom(20)).decode()
```

Comment:
Please follow example from DRF:
https://github.com/encode/django-rest-framework/blob/master/rest_framework/authtoken/models.py
Auth token is generated by a helper function and set on save, which is pretty good. Here we have no option to regenerate permanent token if we want.
And, moreover, the only way to test it is to create object, intead of just check if generate_token function is working.